src/include/duckdb/common/preserved_error.hpp

```diff
@@ -23,8 +23,17 @@ class PreservedError {
 	      exception_instance(nullptr) {
 		// Given a message in the form: 	xxxxx Error: yyyyy
 		// Try to match xxxxxxx with known error so to potentially reconstruct the original error type
-		string err = raw_message.substr(0, raw_message.find(':'));
-		string msg = raw_message.substr(raw_message.find(':') + 2);
+		auto position_semicolon = raw_message.find(':');
```

Comment:
Can we move this code to `PreservedError(const string &raw_message)` (in the .cpp file) and have this constructor only forward to that constructor (i.e. `PreservedError(ex.what())`)?

Reply:
Sure, will do.